src/sap.ui.core/src/sap/ui/core/UIComponent.js

```diff
@@ -479,6 +479,7 @@ sap.ui.define(['jquery.sap.global', '../base/ManagedObject', './Component', './l
 	 * This method has to be overwritten in the implementation of the component
 	 * if the root view is not declared in the component metadata.
 	 *
+	 * @return {sap.ui.core.mvc.View} instance of the view specified in the metadata or in the descriptor file
```

Comment:
createContent might not only return a view, it might return any type of control if it is implemented by a subclass (see getRootControl).